client/extensions/woocommerce/app/order/activity-log/index.js

```diff
@@ -0,0 +1,212 @@
+	} );
+	return eventsByDay;
+}
+
+class ActivityLog extends Component {
+	static propTypes = {
+		isLoaded: PropTypes.bool.isRequired,
+		events: PropTypes.arrayOf( PropTypes.object ).isRequired,
+		days: PropTypes.arrayOf( PropTypes.string ).isRequired,
+		eventsByDay: PropTypes.objectOf( PropTypes.array ).isRequired,
+	}
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			openIndex: 0,
+		};
+	}
+
+	toggleOpenDay = ( index ) => {
+		this.setState( () => ( { openIndex: index } ) );
+	}
+
+	eventPropsByType = {
+		[ EVENT_TYPES.INTERNAL_NOTE ]: ( event ) => {
+			const { translate } = this.props;
+			return {
+				icon: 'aside',
+				// @todo Add comment author once we have that info
+				heading: translate( 'Internal note' ),
+				timestamp: event.timestamp,
+				children: decodeEntities( stripHTML( event.content ) ),
+			};
+		},
+
+		[ EVENT_TYPES.CUSTOMER_NOTE ]: ( event ) => {
+			const { translate } = this.props;
+			return {
+				icon: 'mail',
+				// @todo Add comment author once we have that info
+				heading: translate( 'Note sent to customer' ),
+				timestamp: event.timestamp,
+				children: decodeEntities( stripHTML( event.content ) ),
+			};
+		},
+
+		[ EVENT_TYPES.LABEL_PURCHASED ]: ( event ) => {
+			return {
+				icon: 'print',
+				timestamp: event.timestamp,
+				children: (
+					<LabelItem
+						label={ event }
+						orderId={ this.props.orderId }
+						siteId={ this.props.siteId }
+					/>
+				),
+			};
+		},
+
+		[ EVENT_TYPES.LABEL_REFUND_REQUESTED ]: ( event ) => {
+			return {
+				icon: 'time',
+				timestamp: event.timestamp,
+				children: (
+					<div>
+						<span>Label #{ event.labelIndex + 1 } refund requested</span>
+						{ event.amount != null ? <span> ({ formatCurrency( event.amount, event.currency ) })</span> : null }
+					</div>
+				),
+			};
+		},
+
+		[ EVENT_TYPES.LABEL_REFUND_COMPLETED ]: ( event ) => {
+			return {
+				icon: 'refund',
+				timestamp: event.timestamp,
+				children: (
+					<div>
+						Label #{ event.labelIndex + 1 } refunded ({ formatCurrency( event.amount, event.currency ) })
+					</div>
+				),
+			};
+		},
+
+		[ EVENT_TYPES.LABEL_REFUND_REJECTED ]: ( event ) => {
+			return {
+				icon: 'cross-small',
+				timestamp: event.timestamp,
+				children: (
+					<div>
+						Label #{ event.labelIndex + 1 } refund rejected
+					</div>
+				),
+			};
+		},
+	}
+
+	renderEvents = () => {
+		const { days, eventsByDay, translate } = this.props;
+		if ( ! days.length ) {
+			return (
+				<p>{ translate( 'No activity yet' ) }</p>
+			);
+		}
+
+		return days.map( ( day, index ) => {
+			const events = eventsByDay[ day ];
+			return (
+				<EventsByDay
+					key={ day }
+					count={ events.length }
+					date={ day }
+					index={ index }
+					isOpen={ index === this.state.openIndex }
+					onClick={ this.toggleOpenDay } >
+					{ events.map( event => {
+						const eventProps = this.eventPropsByType[ event.type ]( event );
```

Comment:
I would pass the entire event to `Event`, and pull out the correct props inside that component - it would clean up this component a little to keep that all together.